NICE.Identity.Test/UnitTests/Authorisation.WebAPI/Factories/EmailGeneratorTests.cs

```diff
@@ -0,0 +1,327 @@
+            var expectedSubject = "Dormant account deletion";
+            var expectedHtmlInnerBody = "\r\n" +
+                                        $"{_tabs}Your NICE Account has not been used for 3 years. Unfortunately, we had to delete your details in compliance with GDPR.\r\n" +
+                                        $"{_tabs}<br /><br />\r\n" +
+                                        $"{_tabs}If you want to re-register for a NICE Account using this email address, you will have to re-submit your details on the <a href=\"https://www.nice.org.uk\">NICE website</a>.If you need help you can contact usThis email has been generated automatically. Please do not reply.\r\n";
+            var expectedTextInnerBody = "Your NICE Account has not been used for 3 years. Unfortunately, we had to delete your details in compliance with GDPR.\r\n" +
+                                        "\r\n" +
+                                        "If you want to re-register for a NICE Account using this email address, you will have to re-submit your details on the NICE website.If you need help you can contact usThis email has been generated automatically. Please do not reply.\r\n";
+
+            var expectedHTMLBody = _htmlTemplate.Replace("<%%%TITLE%%%>", expectedTitle).Replace("<%%%BODY%%%>", expectedHtmlInnerBody);
+            var expectedTextBody = _textTemplate.Replace("<%%%TITLE%%%>", expectedTitle).Replace("<%%%BODY%%%>", expectedTextInnerBody);
+
+            email.Subject.ShouldBe(expectedSubject);
+            email.From.Mailboxes.First().Address.ShouldBe(_fromAddress);
+            email.To.Mailboxes.First().Address.ShouldBe(user.EmailAddress);
+            email.HtmlBody.ShouldBe(expectedHTMLBody);
+            email.TextBody.ShouldBe(expectedTextBody);
+
+        }
+
+        [Fact]
+        public void generate_pending_dormant_account_with_roles_removal_notification_email()
+        {
+            //Arrange
+            var context = GetContext();
+            var userService = new UsersService(context, _usersLogger.Object, _providerManagementService.Object, _emailService.Object);
+            var roleService = new RolesService(context, _Roleslogger.Object);
+            var websiteService = new WebsitesService(context, _websiteslogger.Object);
+            var serviceService = new ServicesService(context, _serviceslogger.Object);
+            var environmentService = new EnvironmentsService(context, _environmentslogger.Object);
+
+            var emailGenerationUser = "auth|emailGenerationUser_" + Guid.NewGuid();
+
+            //Create Environment
+            environmentService.CreateEnvironment(new ApiModels.Environment
+            {
+                Name = "Live",
+                Order = 1,
+            });
+            var environment = context.Environments.Single();
+
+            //Create Services
+            serviceService.CreateService(new ApiModels.Service
```

Comment:
comments are superfluous, I know its creating services because the methods called CreateService